src/backend/common/CommandBuffer.cpp

```diff
@@ -201,6 +201,19 @@ namespace backend {
             auto it = mostRecentTextureUsages.find(texture);
             return it != mostRecentTextureUsages.end() && (it->second & usage);
         };
+        auto isTextureTransitionPossible = [&](TextureBase* texture, nxt::TextureUsageBit usage) -> bool {
+            const nxt::TextureUsageBit attachmentUsages =
+                nxt::TextureUsageBit::ColorAttachment |
+                nxt::TextureUsageBit::DepthStencilAttachment;
+            ASSERT(usage != nxt::TextureUsageBit::None && nxt::HasZeroOrOneBits(usage));
+            auto it = mostRecentTextureUsages.find(texture);
```

Comment:
Looks better with the transition into attachments forbidden. We are starting to build a bunch of special cases, if we need to write a spec for NXT I think we'll need to revisit the transitions rule a bit to see if they can be simplified. That said it is fine for now.